api/capi/include/nntrainer.h

```diff
@@ -227,6 +289,64 @@ int ml_nnoptimizer_delete(ml_nnopt_h opt);
  */
 int ml_nnoptimizer_set_property(ml_nnopt_h opt, ...);
 
+/**
+ * @brief Create a dataset with generators to feed to a neural network.
+ * @details Use this function to create a Neural Network Dataset using
+ * generators. The generators will provide data of the given bytes representing
+ * a single input.
+ * @since_tizen 6.x
+ * @param[out] dataset The NNTrainer Dataset handler from the given description.
+ * @param[in] train_cb The dataset generator for training.
+ * @param[in] valid_cb The dataset generator for validating. Can be null.
+ * @param[in] test_cb The dataset generator for testing. Can be null.
+ * @return @c 0 on success. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful.
+ * @retval #ML_ERROR_INVALID_PARAMETER Invalid Parameter.
+ */
+int ml_nndataset_create_with_generator(ml_nndataset_h *dataset,
```

Comment:
Would there be usecases that user doing only testing without training?

Reply:
With the current setup, `train_cb` should be necessarily provided. So, not allowed for now.
Later, can support train_cb to be null to support testing without training.

Reply:
Reasonable.